gui/App.cpp

```diff
@@ -0,0 +1,41 @@
+#include "App.h"
+
+App::App(SDL_Window *w, SDL_Renderer *r) {
+	this->window = w;
+	this->renderer = r;
+}
+
+App::~App() {
+	debug("calling app destructor");
+	for(Widget *widget : widgets) {
+		delete widget;
+	}
+}
+
+App* App::mainloop() {
+	SDL_Event event;
+	while(true) {
+		while(SDL_PollEvent(&event)) {
+			if(event.type == SDL_QUIT) {
+				return nullptr;
+			}
+			for (Widget *widget : this->widgets) {
```

Comment:
(Style) Might want to add a line break between the quit-condition and the rest of the code